backend/tournesol/views/unconnected_entities.py

```diff
@@ -75,7 +87,8 @@ def get_related_entities(entity_id):
         )
 
         return (
-            Entity.objects.filter(id__in=user_all_entities - user_related_entities)
+            Entity.objects.filter(id__in=entity_ids_filter)
+            .annotate(distance=Case(*distance_annotation_whens, default=1+len(neighbors)))
             .annotate(n_comparisons=Subquery(comparison_counts))
-            .order_by("n_comparisons")
+            .order_by("-distance", "n_comparisons")
```

Comment:
The performance of this queryset seems to be okay actually. The bad performance we observed was probably due to heavy/unoptimized test setup :shrug: 

Locally on the dev-env with 1277 compared videos, I get a response from the API in ~200ms.

To be extra cautious, we could filter this queryset to keep (e.g) only the 1000 entities with the largest distance. That would prevent the SQL query from becoming very large (and possibly very slow) for some users in the future, without the need to re-implement the pagination.

Reply:
I went on an optimized it anyway. Customizing the Paginator class was not too difficult. I also added tests that cheeck that the API call takes < 1s.

Reply:
Even the 100'000 videos with ~200'000 comparisons passes the test for < 1s api call (although the setup for the test takes 10 minutes (that's surprisingly slow...)